java-frontend/src/test/java/org/sonar/java/resolve/SymbolTableTest.java

```diff
@@ -98,7 +97,7 @@ public void Generics() {
     JavaSymbol applyMethod = result.symbol("apply");
     assertThat(result.reference(83, 12)).isSameAs(applyMethod);
     // FIXME SONARJAVA-1606 should be 1, subtyping of type variable is wrong
```

Comment:
Accordingly to your modifications, this FIXME should be removed, the 2 remaining usages are perfectly valid. The ones which where wrong are not resolved anymore. Adding the reference to line 89 of Generics file would be great too.